pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -323,6 +330,41 @@ func (asw *actualStateOfWorld) RemoveVolumeFromReportAsAttached(volumeName v1.Un
 	return nil
 }
 
+// Marks the specified volume as being set up
+func (asw *actualStateOfWorld) MarkVolumeAsBeingSetUp(volumeName v1.UniqueVolumeName) {
+	asw.Lock()
+	defer asw.Unlock()
+
+	volumeObj, volumeExists := asw.attachedVolumes[volumeName]
+	if volumeExists {
+		volumeObj.inSetUp = true
+		asw.attachedVolumes[volumeName] = volumeObj
+	}
+}
+
+func (asw *actualStateOfWorld) IsVolumeBeingSetUp(volumeName v1.UniqueVolumeName) bool {
+	asw.RLock()
+	defer asw.RUnlock()
+	volumeObj, volumeExists := asw.attachedVolumes[volumeName]
+	if volumeExists {
+		return volumeObj.inSetUp
+	}
+
+	return false
+}
+
+// Unmarks the specified volume as being set up
+func (asw *actualStateOfWorld) MarkVolumeAsSetUp(volumeName v1.UniqueVolumeName) {
+	asw.Lock()
+	defer asw.Unlock()
+
+	volumeObj, volumeExists := asw.attachedVolumes[volumeName]
+	if volumeExists {
```

Comment:
is it ok to just ignore if volume doesn't exist?

Reply:
dtto. If the volume is not attached, it should be unmounted so the fact it is SetUp doesn't matter.